src/client/src/.tests/components/chatpage.test.js

```diff
@@ -0,0 +1,164 @@
+const React = require('react')
+const ChatPage = require('../../components/chatpage')
+const ConnectStatus = require('../../components/connectStatus')
+const Room = require('../../components/room')
+const {configure, mount, shallow } = require('enzyme')
+const {BrowserRouter} = require('react-router-dom')
+const Adapter = require('enzyme-adapter-react-16')
+const DUMMY_ROOM = 'dummyRoom'
+const DUMMY_USER = 'dummyUser'
+const USERROOMS = 'userRooms'
+const DUMMY_MESSAGE = 'fakeMessage'
+const MESSAGES = 'messages'
+const SIDEBAR = '.sidebar'
+const USERNAME = 'username'
+const MESSAGE_AREA = '.message-area'
+const SELECTED_ROOM = 'selectedRoom'
+const MESSAGE_CLASS = '.message'
+const DUMMY_INPUT = 'dummyInput'
+const CONNECTED = 'connected'
+const INPUT = 'input'
+const HOME_PATH = '/'
+const EXPECTED_ELEMENTS_COUNT = 1
+const ZERO_INDEX = 0
+const EXPECT_NOTHING = 0
+const EMPTY_MAGIC_STRING = ''
+const FALSE = false
+const UNDEFINED = undefined
+/* eslint-disable no-unused-vars */
+const sessionStorage = require('mock-local-storage')
+/* eslint-enable no-unused-vars */
+let suite = {}
+
+global.window = {}
+
+configure({adapter: new Adapter()})
+describe('<ChatPage />', () => {
+	beforeEach(() => {
+		suite.wrapper = mount(<BrowserRouter><ChatPage location={{state: {username: DUMMY_USER}}}/></BrowserRouter>)
```

Comment:
Extract `<BrowserRouter><ChatPage location={{state: {username: DUMMY_USER}}}/></BrowserRouter>` to `const`